k8s: Fix missing status subresource on CiliumNode

christarazi opened the pull request:
This fixes the issue where we fail to update the status of the
CiliumNode resource, leading to failures in surfacing ENI or Azure IPAM
information.

Fixes: https://github.com/cilium/cilium/issues/13193

```release-note
Fix bug in EKS environments where Cilium agents never become ready due to the CiliumNode CRD disallowing updates
```

christarazi committed:
k8s: Fix missing status subresource on CiliumNode

This fixes the issue where we fail to update the status of the
CiliumNode resource, leading to failures in surfacing ENI or Azure IPAM
information.

Fixes: https://github.com/cilium/cilium/issues/13193

Signed-off-by: Chris Tarazi <chris@isovalent.com>

sayboras commented on pkg/k8s/apis/cilium.io/v2/types.go:
```diff
@@ -292,6 +292,7 @@ type CiliumEndpointList struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 // +kubebuilder:resource:singular="ciliumnode",path="ciliumnodes",scope="Cluster",shortName={cn,ciliumn}
 // +kubebuilder:storageversion
+// +kubebuilder:subresource:status
```
I believe that you already scanned once to make sure this is only resource missing `subresource:status` :)

sayboras reviewed (approved):
LGTM

aanm reviewed (changes requested):
Please write a good release note for our users.

qmonnet reviewed (approved):
Thanks a lot for tracking this down!

christarazi committed:
k8s: Mark NodeStatus as optional inside CiliumNode

This field was marked as required incorrectly, leading to failures to
create a CiliumNode (CN) resource.

Seen in Cilium agent logs:

```
2020-09-17T18:52:23.728269658Z level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date"
                               name=CiliumNetworkPolicy/v2 subsys=k8s
2020-09-17T18:52:23.733970534Z level=fatal msg="Unable to create CiliumNode resource"
                               error="CiliumNode.cilium.io \"k8s1\" is invalid: status: Required value"
                               subsys=nodediscovery
```

Signed-off-by: Chris Tarazi <chris@isovalent.com>
